eth2/simulation/src/simulation.rs

```diff
@@ -59,25 +228,242 @@ mod args {
+            initial_state,
+            wasm_code: wasm_code.to_vec(),
+        };
+        let ee_index = simulation
+            .create_execution_environment(create_ee_args)
+            .unwrap();
+        assert_eq!(ee_index, EeIndex::new(0));
+
+        // Set up a shard transaction with the specified data
+        let shard_transaction = ShardTransaction {
+            data: VariableList::new(data).unwrap(),
+            ee_index,
+        };
+        let shard_transaction_copy = shard_transaction.clone();
+
+        // Create a shard block with the one transaction in it
+        let create_shard_block_args = args::CreateShardBlock {
+            shard,
+            shard_transactions: vec![shard_transaction],
+        };
+        // This creates the block and runs all the transactions inside it
+        let shard_slot = simulation
+            .create_shard_block(create_shard_block_args)
+            .unwrap();
+
+        // Get back the EE state to make sure it matches the expected_post_state
+        let get_ee_state_args = args::GetExecutionEnvironmentState { ee_index, shard };
+        let ee_post_state = simulation
+            .get_execution_environment_state(get_ee_state_args)
+            .unwrap();
+        assert_eq!(
+            ee_post_state, expected_post_state,
+            "actual post state root should match expected post state root"
+        );
+
+        (simulation, shard_transaction_copy, shard_slot, ee_index)
+    }
+
+    #[test]
+    fn run_scout_helloworld_and_get_shard_block_and_state() {
+        let initial_state = Root::from([0; 32]);
+        let expected_post_state = Root::from([0; 32]);
+        let data: Vec<u8> = Vec::new();
+        let shard = Shard::new(0);
+        let (simulation, shard_transaction, shard_slot, ee_index) =
+            test_block_with_single_transaction(
+                include_bytes!("../tests/phase2_helloworld.wasm"),
+                initial_state,
+                data,
+                expected_post_state,
+                shard,
+            );
+
+        // Test that GetShardBlock is working as expected
+        let get_shard_block_args = args::GetShardBlock { shard, shard_slot };
+        let shard_block = simulation.get_shard_block(get_shard_block_args).unwrap();
+
+        // Make sure the transaction on the retrieved block matches the transaction
+        // on the created block
+        assert_eq!(
+            shard_block.transactions.get(0).unwrap().clone(),
+            shard_transaction
+        );
+
+        // Test that GetShardState is working as expected
+        let get_shard_state_args = args::GetShardState { shard };
+        let shard_state = simulation.get_shard_state(get_shard_state_args).unwrap();
+        let ee_index: usize = ee_index.into();
+        let ee_state: &Root = shard_state
+            .execution_environment_states
+            .get(ee_index)
+            .unwrap();
+        assert_eq!(ee_state, &expected_post_state);
+    }
```

Comment:
thanks for all the tests - helps see the interface.

Reply:
agreed